sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -495,6 +496,8 @@ object ParquetFileFormat extends Logging {
         } else {
           throw new IOException(s"Could not read footer for file: $currentFile", e)
         }
+      } finally {
+        pool.shutdown()
```

Comment:
I would expect this will fail some test, but it didn't...

When you fix this error, could you call `ThreadUtils.newForkJoinPool` instead to set a better thread name?